### PR TITLE
[WIP] Update Transaction trait and implementations to return counts

### LINKED ISSUE #192: Calling set_edge_properties doesn't return error if edge doesn't exist
There doesn't seem to be any way to know this operation succeeded. Is this really the case?

### PR DESCRIPTION
**Pull request by ozgrakkurt**
Update Transaction trait and implementations to return counts from set and delete operations for checking if operation succeeded.

closes #192

### CODE REVIEW

**Comment by ozgrakkurt** [comment]
@ysimonson Can you review the changes I made to `MemoryDatastore` when you have time?

File: lib/src/memory/datastore.rs [hunk 1]
```diff
@@ -147,58 +147,40 @@ impl InternalMemoryDatastore {
...
     }
 
-    fn delete_vertices(&mut self, vertices: Vec<Uuid>) {
-        for vertex_id in vertices {
-            self.vertices.remove(&vertex_id);
+    fn delete_vertex(&mut self, vertex_id: Uuid) {
+        self.vertices.remove(&vertex_id);
 
-            let mut deletable_vertex_properties: Vec<(Uuid, String)> = Vec::new();
+        let mut deletable_vertex_properties: Vec<(Uuid, String)> = Vec::new();
```
Comment (ysimonson) [inline lib/src/memory/datastore.rs:153]: This isn't used at all

File: lib/src/memory/datastore.rs [hunk 2]
```diff
@@ -147,58 +147,40 @@ impl InternalMemoryDatastore {
...
-            }
-
-            for property_key in deletable_vertex_properties {
-                self.vertex_properties.remove(&property_key);
+            if &vertex_id != property_vertex_id {
+                break;
             }
 
-            let mut deletable_edges: Vec<EdgeKey> = Vec::new();
+            self.vertex_properties.remove(&property_key);
```
Comment (ysimonson) [inline lib/src/memory/datastore.rs:162]: This is iterating through `self.vertex_properties` and removing elements at the same time, which should fail AFAIK

**Comment by ysimonson** [comment]
You can test out specifically these changes by cding into lib and running `cargo test --features=test-suite`. That will also disregard errors in rocksdb and sled not having implemented the trait changes yet.

**Comment by ysimonson** [comment]
@ozgrakkurt how's progress on this? I'm wondering if I should wait for this before releasing 3.0.

**Comment by ozgrakkurt** [comment]
> @ozgrakkurt how's progress on this? I'm wondering if I should wait for this before releasing 3.0.

I hope I can finish it by the end of the week.

**Comment by ysimonson** [comment]
I'm going to close this out since there hasn't been progress in the past few months. Hoping you can circle back to this at some point though @ozgrakkurt!

**Pull request closed without merging**